### PR TITLE
fix: download image removes data about the timer

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- When the user downloaded the image we were not considering that the download status could change, and therefore were not displaying the timer info.

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

image

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1825/1/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt [hunk 1]
```diff
@@ -266,33 +268,46 @@ fun MessageExpireLabel(messageContent: UIMessageContent?, timeLeft: String) {
...
+                statusText = if (messageContent.downloadStatus == Message.DownloadStatus.SAVED_INTERNALLY ||
+                    messageContent.downloadStatus == Message.DownloadStatus.SAVED_EXTERNALLY
+                ) {
+                    stringResource(
+                        R.string.self_deleting_message_time_left,
+                        timeLeft
+                    )
+                } else {
+                    stringResource(R.string.self_deleting_message_label, timeLeft)
+                }
```
Comment (borichellow) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt:296]: IMO `when` looks much better in such cases, but it's up to you

**Review by borichellow — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt [hunk 2]
```diff
@@ -266,33 +268,46 @@ fun MessageExpireLabel(messageContent: UIMessageContent?, timeLeft: String) {
...
         is UIMessageContent.AssetMessage -> {
             StatusBox(
-                statusText = if (messageContent.downloadStatus == Message.DownloadStatus.SAVED_INTERNALLY
-                    || messageContent.downloadStatus == Message.DownloadStatus.SAVED_EXTERNALLY
-                ) stringResource(
-                    R.string.self_deleting_message_time_left,
-                    timeLeft
-                )
-                else stringResource(R.string.self_deleting_message_label, timeLeft)
+                statusText = if (messageContent.downloadStatus == Message.DownloadStatus.SAVED_INTERNALLY ||
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt]: This if is being used in multiple places and if I understand correctly forgetting about SAVED_EXTERNALLY was an issue, so maybe we could create an extension function to make it easier, like `fun DownloadStatus.isSaved(): Boolean`?

**Review by saleniuk — APPROVED** [review]
👌 , just one suggestion

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1825/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1825/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1825/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1825/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6707492)(6707492)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6707492)(6707492)-staging-compat.apk)

**Pull request merged**